common/src/main/java/com/netflix/conductor/common/metadata/workflow/WorkflowDef.java

```diff
@@ -241,6 +241,9 @@ public WorkflowTask getNextTask(String taskReferenceName){
 			 WorkflowTask nextTask = task.next(taskReferenceName, null);
 			 if(nextTask != null){
 				 return nextTask;
+			 } else if (TaskType.DO_WHILE.name().equals(task.getType()) && !task.getTaskReferenceName().equals(taskReferenceName) && task.has(taskReferenceName)) {
```

Comment:
@apanicker-nflx , We need to return null in case all loopOver task execution completed. The issue was, it was returning the next task which was LAMBDA as per [definition](https://github.com/Netflix/conductor/issues/1478#issue-546183053). Since we are returning null it won't change task reference name. LAMBDA task will be return when DO_WHILE task will get completed. And we have added safe guard not to change taskReferenceName in such case.